Polyhedron/demo/Polyhedron/Polyhedron_demo_surface_reconstruction_plugin.cpp

```diff
@@ -0,0 +1,851 @@
+						  
+  }
+
+  void compute_normals (Point_set& points, unsigned int neighbors)
+  {
+    CGAL::jet_estimate_normals<Concurrency_tag>(points.begin(), points.end(),
+                                                CGAL::make_normal_of_point_with_normal_pmap(Point_set::value_type()),
+                                                2 * neighbors);
+    
+    points.erase (CGAL::mst_orient_normals (points.begin(), points.end(),
+					    CGAL::make_normal_of_point_with_normal_pmap(Point_set::value_type()),
+					    2 * neighbors),
+		  points.end ());
+  }
+  
+}
+
+
+
+class Polyhedron_demo_surface_reconstruction_plugin_dialog : public QDialog, private Ui::SurfaceReconstructionDialog
+{
+  Q_OBJECT
+public:
+  Polyhedron_demo_surface_reconstruction_plugin_dialog(QWidget* /*parent*/ = 0)
+  {
+    setupUi(this);
+    
+#ifdef CGAL_EIGEN3_ENABLED
+    m_inputSolver->addItem("Eigen - built-in CG");
+    m_inputSolver->addItem("Eigen - built-in simplicial LDLt");
+#endif
+  }
+
+  unsigned int method () const
+  {
+    if (buttonAuto->isChecked ())       return 0;
+    if (buttonAdvancing->isChecked ())  return 1;
+    if (buttonScaleSpace->isChecked ()) return 2;
+    if (buttonPoisson->isChecked ())    return 3;
+    return -1;
+  }
+  bool boundaries () const { return m_boundaries->isChecked (); }
+  bool interpolate () const { return m_interpolate->isChecked (); }
+  double longest_edge () const { return m_longestEdge->value (); }
+  unsigned int neighbors () const { return m_neighbors->value (); }
+  unsigned int samples () const { return m_samples->value (); }
+  unsigned int iterations () const { return m_iterations->value (); }
+  bool separate_shells () const { return m_genShells->isChecked (); }
+  bool force_manifold () const { return m_forceManifold->isChecked (); }
+  bool generate_smoothed () const { return m_genSmooth->isChecked (); }
+  double angle () const { return m_inputAngle->value (); }
+  double radius () const { return m_inputRadius->value (); }
+  double distance () const { return m_inputDistance->value (); }
+  bool two_passes () const { return m_inputTwoPasses->isChecked (); }
+  bool do_not_fill_holes () const { return m_doNotFillHoles->isChecked (); }
+  QString solver () const { return m_inputSolver->currentText (); }
+};
+
+#include <CGAL/Scale_space_surface_reconstruction_3.h>
+
+class Polyhedron_demo_surface_reconstruction_plugin :
+  public QObject,
+  public CGAL::Three::Polyhedron_demo_plugin_helper
+{
+  Q_OBJECT
+  Q_PLUGIN_METADATA(IID "com.geometryfactory.PolyhedronDemo.PluginInterface/1.0")
+
+  Q_INTERFACES(CGAL::Three::Polyhedron_demo_plugin_interface)
+  QAction* actionScaleSpaceReconstruction;
+
+public:
+  void init(QMainWindow* mainWindow, CGAL::Three::Scene_interface* scene_interface) {
+
+    actionScaleSpaceReconstruction = new QAction(tr("Surface surface reconstruction"), mainWindow);
+    actionScaleSpaceReconstruction->setObjectName("actionSurfaceReconstruction");
```

Comment:
Actually I just meant "Surface reconstruction" (query replace fail...).

Also I noticed that I kept _actionScaleSpaceReconstruction_ as the name of the variable which is not consistent (should be _actionSurfaceReconstruction_).

I'm correcting these right away.